pandas/core/generic.py

```diff
@@ -3082,7 +3082,7 @@ def _box_item_values(self, key, values):
     def _maybe_cache_changed(self, item, value):
         """The object has called back to us saying maybe it has changed.
         """
-        self._data.set(item, value, check=False)
```

Comment:
check is never True, removed the corresponding code in internals.managers and internals.blocks